homeassistant/components/rfxtrx/sensor.py

```diff
@@ -204,6 +204,9 @@ class RfxtrxSensorEntityDescription(SensorEntityDescription):
         state_class=SensorStateClass.MEASUREMENT,
         native_unit_of_measurement=UV_INDEX,
     ),
+    RfxtrxSensorEntityDescription(
```

Comment:
If we do this, it should be disabled by default. But even then, wouldn't it make more sense to make this a trigger and let it feed into a trigger-based template entity?

Reply:
Yes i think so too. It's a rather useless entity on its own. I suppose there could be some benefits from a debugging perspective. Easier to get started, but i do feel just exposing the event seems more logical.

Reply:
It wouldn’t be an event but a trigger. That way we don’t have to post each value to the event bus. So similarly to how MQTT works

Reply:
Ok, so I'm really new to hass, wanted to play around and compare to the openhab I use now, but I need undecoded.

If I understand correctly, you're saying I could just put an event on the bus not associated with a device/entity because this is really going to be picked up by scripts/automations and not used via state. Is that right?

That makes sense, but I have no idea how I would code it. Is it just removing this block and changing the tests or is there more to do? Any pointers on where to look appreciated and I'll give it a go.

Reply:
Reading back, not even an event, directly a trigger?

Reply:
So what we want is to be able to create entities like: 
https://www.home-assistant.io/integrations/template/#trigger-based-template-sensors

They make use of device triggers to sett an entity state based on the data in the payload.

Rfxtrx already expose device triggers for data in an event.  Maybe this is actually already working.

Reply:
Here we have device trigger code: https://github.com/home-assistant/core/blob/dev/homeassistant/components/rfxtrx/device_trigger.py

So you need to extend that a bit for the case of undecoded data.

Reply:
OK, so after implementation the user would need something like this to configure an entity:

```
template:
  trigger:
    - platform: rfxtrx
      type: Undecoded
      subtype: arc
  binary_sensor:
    - state: '{{ ''280ab766'' in trigger.payload }}'
      auto_off: true
```

And that could be a contact switch with the given id so it would trigger when a door is opened, for example.

Or, they could use the same trigger in an automation or similar to do whatever action they wanted without needing to make an entity out of it.

Is that right?

Reply:
Something like that yes. I think this make sense for something where it's fully "custom" data.
If we know the format, it should be parsed somewhere and real entities made.